itou/utils/emails.py

```diff
@@ -69,3 +74,17 @@ def redact_part(part):
     else:
         redacted_domain = redact_part(domain_part)
     return f"{redacted_user}@{redacted_domain}"
+
+
+def check_sender_exists(func):
+    # nearly 2k null senders in DB, mostly prescriber organizations
+    # before fixing the leak, wrapping email sending methods with this
+    # can prevent 500 errors when refusing a job application
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if not args[0].sender:
+            logger.warning(f"send email: job_application:{args[0]} has no 'sender' field")
```

Comment:
Les données variables devraient être logguées séparément.
```suggestion
            logger.warning(f"send email: job_application:%s has no 'sender' field", args[0])
```

Reply:
J'ai lu différentes choses sur le sujet, et à l'occase je veux bien un avis définitif. et ton avis éclairé :) 
Peut-être les parties externes utilisant les logs s'y retrouvent mieux, et je ne suis pas sur que les performances soient tant impactées. De ce que je comprends la notation externe est plus pour de la compatibilité : https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles

Reply:
https://docs.python.org/3/howto/logging.html#logging-variable-data

L’idée, c’est que les formatteurs puissent appliquer les transformations nécessaires aux données variables en inspectant les données variables. Si le message est déjà formatté, cette capacité est perdue.

Reply:
Merci !